### PR TITLE
Added indexer documentation: Reference for Indexing Strategies #78

### PR DESCRIPTION
**Pull request by Sunandhita**

### CODE REVIEW

File: indexer/README.md [hunk 1]
```diff
@@ -1,2 +1,80 @@
 ## Readme for poetry
```
Comment (KaranrajM) [inline indexer/README.md:1]: Remove this line

File: indexer/README.md [hunk 2]
```diff
@@ -1,2 +1,80 @@
...
 
+## Pre processing:
+
+The pre-processing step in indexing involves parsing the file before chunking it. The function parse_file() accepts the entire pathname of the file as a parameter and uses it to extract the extension of the file. Depending on what the file extension is, the corresponding functions for parsing the file are called. Currently, the implementation for file parsing has been provided for files of the format pdf, docs, xslx and json only.
```
Comment (KaranrajM) [inline indexer/README.md:5]: Make the supported file formats BOLD

File: indexer/README.md [hunk 3]
```diff
@@ -1,2 +1,80 @@
...
+
+-**DOCX Parser**: 
+        ```python
+        def docx_parser(docx_file_path: str):
+            return docx2txt.process(docx_file_path)```
```
Comment (KaranrajM) [inline indexer/README.md]: If the code is not needed, can we remove them everywhere?

File: indexer/README.md [hunk 4]
```diff
@@ -1,2 +1,80 @@
...
+
+-**PDF Parser**:
+        ```
+        def pdf_parser(pdf_file_path: str):
+            doc = fitz.open(pdf_file_path)
+            return "\n".join(page.get_text("text") for page in doc)```
+    
+    The parser function pdf_parser() is called when the file to be parsed is a PDF file. This function uses the Fitz python library, which is a part of PyMuPDF package, to open the PDF file specified by the pdf_file_path argument. The `fitz.open()` statement returns a Document object that provides access to the pages and content within the PDF.
+    For every page in the document, the text content of that page is extracted in plain format using get_text("text"). The function, then, joins the text from all pages into one long string with each page's text separated by a newline character (\n)and returns the concatenated text as a string for further processing.
```
Comment (KaranrajM) [inline indexer/README.md:14]: Add a space between (\n)and

File: indexer/README.md [hunk 5]
```diff
@@ -1,2 +1,80 @@
...
+
+-**XLSX Parser**:
+        ```
+        def xlsx_parser(excel_file_path: str):
+            df = pd.read_excel(excel_file_path)
+            return df.to_string(index=False)```
+
+    The parser function xlsx_parser() is called when the file to be parsed is a Excel file. This function uses Pandas python package and DataFrame (a tabular data structure in pandas) for parsing. 
+    `pandas.read_excel()` reads the Excel file provided (excel_file_path) into a DataFrame. By default, `pd.read_excel()` reads the first sheet of the Excel file. After reading, `df.to_string(index=False)` converts the DataFrame into a plain-text string.The result is a formatted string that represents the contents of the Excel sheet in a tabular layout, with columns and rows formatted as text.
```
Comment (KaranrajM) [inline indexer/README.md:19]: Add a space between string.The

File: indexer/README.md [hunk 6]
```diff
@@ -1,2 +1,80 @@
...
+The content collected from each file is chunked in such a way that:
+
+- Every chunk has a maximum of 4000 characters.
+- The first **200 characters overlap** with the last 200 characters of the previous chunk.
+- No sentences are cut in between chunks.
+
+This results in continuous and logically structured chunks for efficient searching and retrieval.
+
+
+## API and Indexer Service Workflow
```
Comment (KaranrajM) [inline indexer/README.md:58]: Remove everything in these ## API and Indexer Service Workflow I will add it

File: indexer/README.md (hunk 1 above)
Comment (Sunandhita) [reply indexer/README.md]: Done.

File: indexer/README.md (hunk 2 above)
Comment (Sunandhita) [reply indexer/README.md]: Done

File: indexer/README.md (hunk 3 above)
Comment (Sunandhita) [reply indexer/README.md]: Done

File: indexer/README.md (hunk 4 above)
Comment (Sunandhita) [reply indexer/README.md]: Done

File: indexer/README.md (hunk 5 above)
Comment (Sunandhita) [reply indexer/README.md]: Done

File: indexer/README.md (hunk 6 above)
Comment (Sunandhita) [reply indexer/README.md:56]: Done

**Pull request merged**